log4j-1.2-api/src/main/java/org/apache/log4j/xml/DOMConfigurator.java

```diff
@@ -66,20 +69,34 @@
  */
 public class DOMConfigurator {
 
+    private static boolean isFullCompatibilityEnabled() {
+        return PropertiesUtil.getProperties().getBooleanProperty(ConfigurationFactory.LOG4J1_EXPERIMENTAL);
+    }
+
+    private static void warnFullCompatibilityDisabled() {
+        LogLog.warn(
+                "Ignoring `DOMConfigurator` call, since `log4j1.compatibility` is not enabled.\n"
+                        + "See https://logging.staged.apache.org/log4j/2.x/migrate-from-log4j1.html#log4j1.compatibility for details.");
+    }
```

Comment:
:bulb: Idea: Instead of these two methods, you could have only introduced a wrapper method as follows:

```
private static void whenFullCompatibilityEnabled(ThrowingRunnable runnable) {
    // ...
}
```

Reply:
Sure, but is it worth introducing `ThrowingRunnable`?

Reply:
@ppkarwasz, I'm leaving the decision to you. I am fine with either approaches.